o1vm/src/mips/tests.rs

```diff
@@ -236,6 +236,31 @@ mod unit {
         env.memory[page as usize].1[page_address + 3] = (instr & 0xFF) as u8;
     }
 
+    pub(crate) fn sign_extend(x: u32, bitlength: u32) -> u32 {
```

Comment:
Discussed on Slack. It seems that this PR tests helpers that you will use later in the code, and it is not the ones in the environment you want to test. Accepting based on this assumption.